ee/allowedcmd/cmd.go

```diff
@@ -13,17 +13,59 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/traces"
 )
 
-type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+type AllowedCommand func(ctx context.Context, arg ...string) (*TracedCmd, error)
+
+type TracedCmd struct {
+	Ctx context.Context // nolint:containedctx // This is an approved usage of context for short lived cmd
+	*exec.Cmd
+}
+
+// Start overrides the Start method to add tracing before executing the command.
+func (t *TracedCmd) Start() error {
+	_, span := traces.StartSpan(t.Ctx, "path", t.Cmd.Path, "args", fmt.Sprintf("%+v", t.Cmd.Args))
+	defer span.End()
+
+	return t.Cmd.Start() //nolint:forbidigo // This is our approved usage of t.Cmd.Start()
+}
+
+// Run overrides the Run method to add tracing before running the command.
+func (t *TracedCmd) Run() error {
+	_, span := traces.StartSpan(t.Ctx, "path", t.Cmd.Path, "args", fmt.Sprintf("%+v", t.Cmd.Args))
+	defer span.End()
+
+	return t.Cmd.Run() //nolint:forbidigo // This is our approved usage of t.Cmd.Start()
```

Comment:
Nitpick for the comment --

```suggestion
	return t.Cmd.Run() //nolint:forbidigo // This is our approved usage of t.Cmd.Run()
```

Reply:
fixed all the comments